### ISSUE TITLE
False positive with compound Boolean expression

### ISSUE THREAD

**Issue by datawookie** [astral-sh/ty#658]
### Summary

Hi!

I'm super impressed by `ty`. The speed is just phenomenal.

I have encountered a surprising result though:

```
warning[possibly-unbound-attribute]: Attribute `month` on type `datetime | None` is possibly unbound
   |
52 |     assert job.posted_at is not None and job.posted_at.month == 5
   |                                          ^^^^^^^^^^^^^^^^^^^
   |
info: rule `possibly-unbound-attribute` is enabled by default
```

I believe that this is a false positive because the first part of the expression rules out the possibility of `job.posted_at` being `NULL`.

Thanks for the great work on this, making the lives of developers everywhere better!

Best regards, Andrew.

**Comment by AlexWaygood** [comment]
Thanks for the report, and for the kind words! Unfortunately this is a duplicate of #164 :-)

**Comment by datawookie** [comment]
Ah! I'm annoyed that I did not find that. Oh well, glad that this is on the radar anyway.
